### DISCUSSION TITLE
the compressor does not work ? Just a gain ?

### DISCUSSION THREAD

**Discussion by Qui-Gon-JooL** [SynthstromAudible/DelugeFirmware#1595] [Q&A] [answered]
Hi,
   its me or is the compressor not working? In its first versions it worked, it reduced the peaks, the various parameters changed things. 

But since some versions, I have the impression that apart from the treshold which is content to add gain all the other parameters do not change absolutely nothing... This is not the behavior of a compressor, a simple gain parameter would be enough in this case...

 Please, can someone look into this and tell me if I'm crazy?

**Comment by sapphire-arches** [comment]
Can you post a song file or an example audio clip demonstrating the change? The compressor implementation was switched to a more efficient and less colorful algorithm in the 1.0 cycle, so maybe that's what you're hearing?

**Comment by Qui-Gon-JooL** [comment]
Sorry, I don't have an internet host to post videos or sound or images, but I have carried out several measurements on ableton live and apart from boosting the volume regardless of the settings, nothing happens.

 whether on the master or on a kit... impossible to either attenuate the peaks or add punch to the attacks... This has absolutely no effect apart from increasing the volume, once returned to equal volume absolutely nothing difference.

and this even with a treshold and an extreme ratio. I really think there is a big problem.
and it's really a shame because I remember having an effective compressor on a nightly which was 2 or 3 months old...

**Reply by sapphire-arches** [reply]
You can upload files directly to this thread on GitHub. It's possible we have a bug.

**Comment by Qui-Gon-JooL** [comment]
(I can't put sound in anything other than an archive?...)

here are 2 audio examples:

 in the first example (Test01) there are 3 loops placed end to end:
 
[Test01.zip](https://github.com/SynthstromAudible/DelugeFirmware/files/14782545/Test01.zip)

- 00:00 to 00:16 the sound as is without compressor.

- 00:16 to 00:32 a compressor set treshold and ratio fully with an attack at 15 and release at 110 which should let the transiants through and lower the volume behind the transiants.

- 00:32 to 00:48 a compressor set treshold and ratio fully with an attack at 0 and release at 110 which should cut the peaks.

what I notice is a clear increase in volume on the second and third loop but not the expected compressor effect...

in the second example (Test02) it's the same thing, I just reduced the volume of loops two and three (the compressed loops) once in the audio editor so that it corresponds to the volume of the first loop.

[Test02.zip](https://github.com/SynthstromAudible/DelugeFirmware/files/14782578/Test02.zip)

**Comment by Qui-Gon-JooL** [comment]
there is indeed a slight action but very confusing on the full parameters... and even treshold and ratio at max. on loops 2 and 3 it peaks a little less strong at equal lufs but even if I let some signal pass by increasing the attack which is not logical...

I have the impression that only the treshold button works and that it acts as an on\off on a type of non-configurable limiter\maximizer...

**Comment by m-m-adams** [comment]
Are you comparing to version 1.0? If it's changed then it's a bug

**Comment by Qui-Gon-JooL** [comment]
No, I'm talking about the nlghtly versions here, all for about 2 months, in fact I found the compressor effective until the compressor by track and kit appeared, from there, the behavior of the compressors including the master has changed, has become what it is...

**Reply by m-m-adams** [reply]
Merged some fixes, give it a try tomorrow but it's back to how it was working pre February. Just a couple bugs that added up to the compressor not doing anything except add gain

**Reply by Qui-Gon-JooL** [reply]
ok, thank you, I tried the update last night, it seems to work better but now there is a serious problem with the general volume too low, especially the kits...

**Comment by m-m-adams** [comment, marked as answer]
Yeah confirmed, I think the volume adjustment is off. It's pretty noticeable back to back from 1.0 to current nightly so not sure how I missed it
